mpas_analysis/shared/timekeeping/Date.py

```diff
@@ -75,6 +76,50 @@ def __init__(self, dateString=None, isInterval=False, totalSeconds=None,
             self.seconds = numpy.int64(seconds)
             self._setTotalSeconds()
 
+    def to_datetime(self, yearOffset=0):
+        """
+        Converts the date object to a datetime object.
+        The yearOffset is added to this date's year, and
+        the resulting date is clamped to the range supported by
+        numpy's datetime64[ns], used internally by xarray an
+        pandas
+
+        Last modified: 11/28/2016
+        Author: Xylar Asay-Davis
+        """
+        if self.isInterval:
+            raise ValueError("self.isInterval == True. Use to_timedelta "
+                             "instead of to_datetime")
+
+        year = numpy.maximum(datetime.MINYEAR,
+                             numpy.minimum(datetime.MAXYEAR,
+                                           self.years+yearOffset))
+        outDate =  datetime.datetime(year=year, month=self.months+1,
+                                     day=self.days+1, hour=self.hours,
+                                     minute=self.minutes, second=self.seconds)
+
+        minDate = datetime.datetime(year=1678, month=1, day=1,
+                                    hour=0, minute=0, second=0)
+        maxDate = datetime.datetime(year=2262, month=1, day=1,
+                                    hour=0, minute=0, second=0)
+        outDate = max(minDate, min(maxDate, outDate))
```

Comment:
I disagree that this should be a warning.  Typically want the end year of the time series analysis to be `9999` so we don't have to think about it.  This is the function that makes sure that `9999 + 1849` still gives you a reasonable year that the analysis can handle.  If you didn't want the clamping and the year offset, you wouldn't bother calling this function.  

If I add a warning, it's going to be annoying that there are warnings throughout our output even though I'm intentionally using this function to clamp the date. The whole *point* of clamping the date here is so I don't have to do it elsewhere and so I can have better code reuse.